### PR TITLE
Fix skip bug when using include/require controls.

### PR DESCRIPTION
**Pull request by jquick**
Signed-off-by: Jared Quick <jquick@chef.io>

Currently the new skip code will still generate controls inside include/require block. This fixes that issue.

Fixes https://github.com/inspec/inspec/issues/3486

### CODE REVIEW

File: test/unit/mock/profiles/unsupported_dependencies/inspec.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: unsupported_dependencies
+title: InSpec Profile
+maintainer: The Authors
+copyright: The Authors
+copyright_email: you@example.com
+license: Apache-2.0
+summary: An InSpec Compliance Profile
```
Comment (clintoncwolfe) [inline test/unit/mock/profiles/unsupported_dependencies/inspec.yml]: Please give a purpose for this fixture profile.  We have many, and an understanding of what this is for will help your fellow developers.

File: test/unit/mock/profiles/unsupported_dependencies/inspec.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+version: 0.1.0
+depends:
+  - name: child_profile
+    url: https://example.com/child_profile.tar.gz
```
Comment (clintoncwolfe) [inline test/unit/mock/profiles/unsupported_dependencies/inspec.yml]: Consider using a `path` here instead, and pointing to the other fixture profile.

File: test/unit/mock/profiles/unsupported_dependencies/vendor/e39eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/inspec.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: child_profile
+title: Child Profile
+maintainer: Demo, Inc.
+copyright: Demo, Inc.
+copyright_email: support@example.com
+license: Apache-2.0
+summary: My Profile 1 summary
```
Comment (clintoncwolfe) [inline test/unit/mock/profiles/unsupported_dependencies/vendor/e39eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/inspec.yml]: Please provide a meaningful summary

File: test/unit/mock/profiles/unsupported_dependencies/vendor/f24eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/inspec.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+name: child_profile2
+title: Child Profile2
+maintainer: Demo, Inc.
+copyright: Demo, Inc.
+copyright_email: support@example.com
+license: Apache-2.0
+summary: My Profile 1 summary
```
Comment (clintoncwolfe) [inline test/unit/mock/profiles/unsupported_dependencies/vendor/f24eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/inspec.yml]: Please provide a meaningful summary

File: test/unit/mock/profiles/unsupported_dependencies/vendor/f24eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/controls/profile1.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+control 'include-01' do
```
Comment (clintoncwolfe) [inline test/unit/mock/profiles/unsupported_dependencies/vendor/f24eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/controls/profile1.rb]: Comments here explaining how this fixture is intended to be used would be helpful.

File: test/unit/mock/profiles/unsupported_dependencies/vendor/e39eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/controls/profile1.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+control 'require-01' do
```
Comment (clintoncwolfe) [inline test/unit/mock/profiles/unsupported_dependencies/vendor/e39eb85366b272bae98e5eecdfac9f84c50a9ae9dd625fba2ce847268a6c3477/controls/profile1.rb]: Comments here explaining how this fixture is intended to be used would be helpful.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
A few requested changes to the way the test fixtures are handled.  Code and test itself looks good!

**Comment by jquick** [comment]
Thanks @clintoncwolfe  should be set

**Comment by jquick** [comment]
The only failing test is the inspec version is out of date. Branch `inspec-3.0` needs refreshed for that.

**Review by miah — APPROVED** [review]
Thanks for finding that @jquick!

**Review by clintoncwolfe — APPROVED** [review]
Looks good, thanks!

**Pull request merged**